### DISCUSSION TITLE
Refining CrewAI Results Through Chat Interaction

### DISCUSSION THREAD

**Discussion by ItouTerukazu** [crewAIInc/crewAI#1329] [General]
I'm curious about how others are implementing a chat-based system to refine the results obtained from CrewAI crews. Specifically, I'm interested in approaches that allow users to provide feedback like "I'd like this part to be a bit different" and then iteratively improve the crew's output through a conversational interface.
How are you handling this kind of interactive refinement process in your projects? Any advice or best practices would be greatly appreciated.
Thank you in advance for your insights!

**Comment by gadgethome** [comment]
We do have the new agent training feature which  really helps.  Also we are working on improving the human in the loop. Thanks
